[2020-12-22 14:05:30,440] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:05:30,461] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:05:30,463] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:05:30,464] {taskinstance.py:867} INFO - Starting attempt 38 of 42
[2020-12-22 14:05:30,464] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:05:30,478] {taskinstance.py:887} INFO - Executing <Task(LivyBatchOperator): preprocess_data> on 2020-12-22T12:43:23.210460+00:00
[2020-12-22 14:05:30,482] {standard_task_runner.py:53} INFO - Started process 56642 to run task
[2020-12-22 14:05:30,548] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [running]> dcb43f18a24b
[2020-12-22 14:05:30,574] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,573] {batch.py:214} INFO - Submitting the batch to Livy... Payload:
{
  "file": "s3a://alanchn31-loan-default-prediction/dist/main.py",
  "args": [
    "--job",
    "preprocess_data",
    "--phase",
    "train",
    "--mode",
    "aws",
    "--awsKey",
    "AKIAUW4FKHTMR2UR3INH",
    "--awsSecretKey",
    "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",
    "--s3Bucket",
    "alanchn31-loan-default-prediction"
  ],
  "pyFiles": [
    "s3a://alanchn31-loan-default-prediction/dist/src.zip"
  ],
  "name": "preprocess_data_scheduled__2020-12-22T12:43:23.210460+00:00"
}
[2020-12-22 14:05:30,584] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,583] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:05:30,587] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,587] {http_hook.py:136} INFO - Sending 'POST' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches
[2020-12-22 14:05:30,609] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,609] {batch.py:166} INFO - Batch successfully submitted with id = 34.
[2020-12-22 14:05:30,612] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,612] {batch.py:80} INFO - Getting batch 34 status...
[2020-12-22 14:05:30,620] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,620] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:05:30,623] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,623] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/34
[2020-12-22 14:05:30,635] {logging_mixin.py:112} INFO - [2020-12-22 14:05:30,635] {batch.py:90} INFO - Batch 34 has not finished yet (state is 'starting')
[2020-12-22 14:05:50,654] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,654] {batch.py:80} INFO - Getting batch 34 status...
[2020-12-22 14:05:50,661] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,661] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:05:50,663] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,663] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/34
[2020-12-22 14:05:50,671] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,670] {batch.py:299} INFO - --------------------------------------------------Full log for batch 34--------------------------------------------------
[2020-12-22 14:05:50,679] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,678] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:05:50,682] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,682] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/34/log?from=0&size=100
[2020-12-22 14:05:50,689] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,689] {batch.py:309} INFO - stdout: 
[2020-12-22 14:05:50,690] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,690] {batch.py:309} INFO - 2020-12-22 14:05:32,393 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2020-12-22 14:05:50,690] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,690] {batch.py:309} INFO - Traceback (most recent call last):
[2020-12-22 14:05:50,690] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,690] {batch.py:309} INFO -   File "/tmp/spark-f79f724b-2d9a-4f2d-ba5d-9affe2bae7c6/main.py", line 86, in <module>
[2020-12-22 14:05:50,691] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,690] {batch.py:309} INFO -     main()
[2020-12-22 14:05:50,691] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,691] {batch.py:309} INFO -   File "/tmp/spark-f79f724b-2d9a-4f2d-ba5d-9affe2bae7c6/main.py", line 56, in main
[2020-12-22 14:05:50,691] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,691] {batch.py:309} INFO -     spark_files_dir = SparkFiles.getRootDirectory()
[2020-12-22 14:05:50,691] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,691] {batch.py:309} INFO -   File "/usr/spark/python/lib/pyspark.zip/pyspark/files.py", line 59, in getRootDirectory
[2020-12-22 14:05:50,692] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,691] {batch.py:309} INFO - AttributeError: 'NoneType' object has no attribute '_jvm'
[2020-12-22 14:05:50,692] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,692] {batch.py:309} INFO - 2020-12-22 14:05:35,305 INFO util.ShutdownHookManager: Shutdown hook called
[2020-12-22 14:05:50,692] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,692] {batch.py:309} INFO - 2020-12-22 14:05:35,307 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-f79f724b-2d9a-4f2d-ba5d-9affe2bae7c6
[2020-12-22 14:05:50,692] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,692] {batch.py:309} INFO - 2020-12-22 14:05:35,313 INFO impl.MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2020-12-22 14:05:50,692] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,692] {batch.py:309} INFO - 2020-12-22 14:05:35,313 INFO impl.MetricsSystemImpl: s3a-file-system metrics system stopped.
[2020-12-22 14:05:50,693] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,693] {batch.py:309} INFO - 2020-12-22 14:05:35,314 INFO impl.MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2020-12-22 14:05:50,693] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,693] {batch.py:309} INFO - 
stderr: 
[2020-12-22 14:05:50,693] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,693] {batch.py:309} INFO - 
YARN Diagnostics: 
[2020-12-22 14:05:50,694] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,693] {batch.py:309} INFO - spark-submit start failed
[2020-12-22 14:05:50,694] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,694] {batch.py:318} INFO - --------------------------------------------------End of full log for batch 34--------------------------------------------------
[2020-12-22 14:05:50,694] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,694] {batch.py:335} INFO - Closing batch with id = 34
[2020-12-22 14:05:50,701] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,701] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:05:50,703] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,703] {http_hook.py:136} INFO - Sending 'DELETE' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/34
[2020-12-22 14:05:50,712] {logging_mixin.py:112} INFO - [2020-12-22 14:05:50,712] {batch.py:340} INFO - Batch 34 has been closed
[2020-12-22 14:05:50,722] {taskinstance.py:1128} ERROR - Batch 34 failed with state 'dead'
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 966, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 173, in execute
    ).execute(context)
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/sensors/base_sensor_operator.py", line 107, in execute
    while not self.poke(context):
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 96, in poke
    raise AirflowException(f"Batch {self.batch_id} failed with state '{state}'")
airflow.exceptions.AirflowException: Batch 34 failed with state 'dead'
[2020-12-22 14:05:50,724] {taskinstance.py:1151} INFO - Marking task as UP_FOR_RETRY
[2020-12-22 14:05:55,490] {logging_mixin.py:112} INFO - [2020-12-22 14:05:55,490] {local_task_job.py:103} INFO - Task exited with return code 1
